src/h5web/visualizations/nexus/hooks.ts

```diff
@@ -59,29 +54,29 @@ export function useNxData(group: HDF5Group, metadata: HDF5Metadata): NxData {
   const silxStyle = parseSilxStyleAttribute(group);
   const { axes_scale_type, signal_scale_type } = silxStyle;
 
-  const axisMapping = axesNames.map((name, i) => {
-    if (!name) {
+  const axisMapping = axesNames.map((axisName, i) => {
+    if (!axisName) {
       return undefined;
     }
 
-    const dataset = getLinkedEntity(name, group, metadata);
-    if (!dataset) {
-      return undefined;
-    }
-    assertDataset(dataset);
+    const axisDataset = getLinkedEntity(axisName, group, metadata);
+    assertDefined(axisDataset);
```

Comment:
Now enforcing the presence of the axis dataset if it's listed in the `axes` attribute.